objects/src/accounts/storage/header.rs

```diff
@@ -0,0 +1,109 @@
+use alloc::vec::Vec;
+
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{AccountStorage, StorageSlotType, Word};
+
+// ACCOUNT STORAGE HEADER
+// ================================================================================================
+
+/// Account storage header is a lighter version of the [AccountStorage] storing
+/// only the [StorageSlotType] and associated values of the [super::StorageSlot]s
+/// contained in the storage.
+///
+/// The use of a header is useful in the situation where the storage is heavy (i.g. multiple Mb's),
+/// and should be enough to execute most transactions.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountStorageHeader {
+    slots: Vec<(StorageSlotType, Word)>,
+}
+
+impl AccountStorageHeader {
+    /// Returns a reference to the storage header slots.
+    pub fn slots(&self) -> &Vec<(StorageSlotType, Word)> {
+        &self.slots
+    }
```

Comment:
I would probably change the return type of this to `impl Iterator<Item = (StorageSlotType, &Word)>`. This way, we don't expose internal details of the implementation.

Reply:
Yes I can change the return type. 

Can you tell me more on why it would be better to return an iterator over the type than a Vec of the type? 

Thank you.

Reply:
`Vec` is an internal implementation detail. If we decide to change it in the future, it would break the public interface. By returning an iterator we have more flexibility to change internal implementation in the future without breaking things downstream.